packages/cli/src/commands/repo/test.ts

```diff
@@ -15,10 +15,88 @@
  */
 
 import os from 'os';
+import crypto from 'node:crypto';
+import fs from 'fs-extra';
+import yargs from 'yargs';
+import { resolve as resolvePath, relative as relativePath } from 'path';
 import { Command, OptionValues } from 'commander';
-import { PackageGraph } from '@backstage/cli-node';
+import { Lockfile, PackageGraph } from '@backstage/cli-node';
 import { paths } from '../../lib/paths';
-import { runCheck } from '../../lib/run';
+import { runCheck, runPlain } from '../../lib/run';
+import { isChildPath } from '@backstage/cli-common';
+
+type JestProject = {
+  displayName: string;
+};
+
+interface GlobalWithCache extends Global {
+  __backstageCli_jestSuccessCache?: {
+    filterConfigs(
+      projectConfigs: JestProject[],
+      globalConfig: unknown,
+    ): Promise<JestProject[]>;
+    reportResults(results: {
+      testResults: Array<{
+        displayName?: { name: string };
```

Comment:
no existing TS types to rely on for this object? it's mirroring some jest internals isn't it, or am i misunderstanding?

Reply:
Yeh nah, requires installation of `jest-test-result` which is a completely new package and kinda defeats the purpose because it can easily get out of sync